lang/syn/src/codegen/accounts/constraints.rs

```diff
@@ -320,6 +325,43 @@ pub fn generate_constraint_rent_exempt(
     }
 }
 
+fn generate_constraint_realloc(f: &Field, c: &ConstraintReallocGroup) -> proc_macro2::TokenStream {
+    let field = &f.ident;
+    let new_space = &c.space;
+    let allocator = &c.allocator;
+
+    quote! {
+        let __anchor_rent = Rent::get()?;
+
+        let (__delta_space, __additive) = match #new_space.checked_sub(#field.to_account_info().data_len()) {
```

Comment:
This is just `abs_diff` and `>`. No need to `unwrap`.

Reply:
not sure I know what you're trying to say. the `checked_XXX` arithmetic functions return an `Option<T>` type that is `None` on resulting under or overflows. this `match` clause is in place to check for that underflow which would be an indication that the reallocation is substractive and then safely recalculate the value of the difference in byte length.

Reply:
anchor is pinned to rust `1.56` and `abs_diff` was introduced in version `1.60.0`